### PR TITLE
Adding JSON parse response rescue

### PR DESCRIPTION
**Pull request by prpetten**
Adding rescue of 404 on resource.find to return nil

### CODE REVIEW

File: lib/name_drop/resources/base.rb [hunk 1]
```diff
@@ -58,6 +58,9 @@ def self.find(client, id)
         else
           response
         end
+      rescue NameDrop::Error => error
+        return if error.detail['code'] == 404
```
Comment (ridiculous) [inline lib/name_drop/resources/base.rb:62]: Would be cool if we could just rescue from `NameDrop::NotFoundError`

File: lib/name_drop/client.rb [hunk 2]
```diff
@@ -89,10 +89,15 @@ def delete(endpoint)
     # @return [Hash]
     # @raise [NameDrop::Error] Rescuing RestClient::ExceptionWithResponse
     def request(method, endpoint, attributes = {})
-      response = RestClient::Request.execute(create_request_hash(method, endpoint, attributes))
-      JSON.parse(response) unless response.empty?
+      safe_json_parse(RestClient::Request.execute(create_request_hash(method, endpoint, attributes)))
     rescue RestClient::ExceptionWithResponse => err
-      raise NameDrop::Error.new(error_message(method), JSON.parse(err.response).with_indifferent_access)
+      raise NameDrop::Error.new(error_message(method), safe_json_parse(err.response))
```
Comment (ridiculous) [inline lib/name_drop/client.rb:94]: Maybe we could raise a specialized error here?

**Review by ridiculous — COMMENTED** [review]
Thoughts on raising a new error

File: lib/name_drop/client.rb (hunk 2 above)
Comment (prpetten) [reply lib/name_drop/client.rb:94]: We would need to check the response code here for 404, as there are lots of other errors that could be returned from the request, and everywhere else, we wouldn't want to check for 404, just pass it along, the only place we want to check for 404 is in the find.

File: lib/name_drop/resources/base.rb (hunk 1 above)
Comment (prpetten) [reply lib/name_drop/resources/base.rb:62]: This appears to be the appropriate place to do this given that other requests don't need to parse out 404 errors.

**Pull request merged**